### PR TITLE
fix: Form.item not recognized to single form node caused by React empty Node

### LINKED ISSUE #42092: 表单有一个很低级的错误
### Reproduction link

[http://xxxx](http://xxxx)

### Steps to reproduce

```
import React                 from 'react';
import {Button, Form, Input} from 'antd';

function AddAdmin() {
  // const [form] = Form.useForm();

  const onFinish = (values) => {
    console.log(values);
  };

  return (
    <Form name = "basic" autoComplete = "off" onFinish = {onFinish}>
      <Form.Item label = "User" name = "arqr"
                 rules = {[{required: true, message: '请输入数据'},]}
      > <Input/>
      </Form.Item>
      <Form.Item> <Button type = "primary" htmlType = "submit">提交</Button> </Form.Item>
    </Form>
  );
}
```

export default AddAdmin;
使用这个组件，会发现多次输入，表单无法得到新值！
但是如果我把源码中Input前的空格删除，即是>与<之间的空格，删除，就会正常！！

### What is expected?

点提交，表单每次都能获得Input输入的值

### What is actually happening?

多次输入，点提交，表单无法得到新值！

| Environment | Info |
| --- | --- |
| antd | 5.4.6 |
| React | 18 |
| System | win 2012 r2 |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by BoyYangzai**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
- fix #42092 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    fix: Form.item not recognized to single form node caused by React empty Node     |
| 🇨🇳 Chinese |      修复React空节点导致的Form.item无法识别到单一表单节点     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at d44d3f0</samp>

Fixed a bug in `Form` component where empty string children would render an extra row. Added a `filterChildrenEmptyString` function in `components/form/hooks/useChildren.ts` to filter out empty string children.

### <samp>🤖 Generated by Copilot at d44d3f0</samp>

*  Add `filterChildrenEmptyString` function to filter out empty string children from the `childList` array and fix a bug where empty string children would cause the form to render an extra empty row ([link](https://github.com/ant-design/ant-design/pull/42099/files?diff=unified&w=0#diff-bcff919efcf0dcb42c694395467c0b7700162e1ffbfda4f7092b06e53f8ab60fL10-R18))

### CODE REVIEW

File: components/form/hooks/useChildren.ts [hunk 1]
```diff
@@ -7,7 +7,13 @@ export default function useChildren(
...
+  return childList!.length <= 1 ? childList![0] : childList;
+}
 
-  const childList = toArray(children);
-  return childList.length <= 1 ? childList[0] : childList;
+function filterChildrenEmptyString(childrenArray?: Array<any>) {
+  return childrenArray?.filter((item) => {
+    if (typeof item !== 'string') return true;
+    return item && item.trim();
+  });
```
Comment (afc163) [inline components/form/hooks/useChildren.ts:18]: 是不是改到 rc-field-form 里去更好？

Comment (BoyYangzai) [reply components/form/hooks/useChildren.ts:18]: 收到🫡

**Review by Yuiai01 — APPROVED** [review]
LGTM

**Comment by afc163** [comment]
ci 挂了，看一下。

**Comment by yoyo837** [comment]
不好判断是不是用户特意为之吧? 个人觉得是won't fix.

File: components/form/hooks/useChildren.ts [hunk 2]
```diff
@@ -1,13 +1,19 @@
 import toArray from 'rc-util/lib/Children/toArray';
 import type { FormItemProps } from '../FormItem';
 
+function filterChildrenEmptyString(childrenArray?: Array<any>) {
+  return childrenArray?.filter((item) => {
+    if (typeof item !== 'string') return true;
+    return item && item.trim();
+  });
```
Comment (afc163) [inline components/form/hooks/useChildren.ts:8]: 可以考虑 `<Form.Item><Input />,</Form.Item>` `<Form.Item><Input /> 米</Form.Item>` 这种场景。

Comment (MadCcc) [reply components/form/hooks/useChildren.ts:8]: 有文本节点感觉就已经是误用了，我感觉多个逗号，多个空格，甚至多一句话都是一码事

Comment (BoyYangzai) [reply components/form/hooks/useChildren.ts:8]: @afc163 
文本节点应该算是错误使用 该pr应该只处理由React识别Node导致出现的错误空节点这一特殊情况
eg：
image

其余文本节点 类似于 + ， 还是应该和以前的逻辑一样 给予开发者waring较为合适 规范开发者使用 `Single ItemNode`

**Comment by yoyo837** [comment]
Close by https://github.com/ant-design/ant-design/issues/42092#issuecomment-1537674841

File: components/form/hooks/useChildren.ts (hunk 2 above)
Comment (Midas-Li) [reply components/form/hooks/useChildren.ts:8]: 我注重这个空格问题，是不是修复了？如何更新。antd重新npm un再npm i 就可以更新了？

Comment (BoyYangzai) [reply components/form/hooks/useChildren.ts:8]: @Midas-Li 这个pr就是用来修复这个问题的 不过目前已经close了 不会修复了～
辛苦规范下开发代码风格了～

Comment (Midas-Li) [reply components/form/hooks/useChildren.ts:8]: 好吧，不管怎样还是谢谢你。

**Pull request closed without merging**